keps/sig-cloud-provider/20190125-out-of-tree-ibm.md

```diff
@@ -40,60 +43,135 @@ status: provisional
+* Re-architecting the IBM cloud provider beyond transition from IBM KCM to IBM CCM.
+* Open source other components of IKS.
 
 ## Proposal
 
-### Implementation Details/Notes/Constraints [optional]
-
-TODO for SIG-IBM
+### Implementation Details/Notes/Constraints
+
+IBM has already completed most of the IBM CCM enablement work. Furthermore, IBM
+has obtained the necessary internal approvals to open source the IBM cloud
+provider. The remaining implementation details covered by this KEP are for
+IBM CCM implementation, test and documentation.
+
+IBM KCM only implements the node and service controllers (router controller
+is handled by a [Calico](https://www.projectcalico.org/) overlay). These
+components are ready for the switch to the IBM CCM per
+[KEP-20180530-cloud-controller-manager](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20180530-cloud-controller-manager.md).
+This leaves implementation of the actual CCM command along with development of a
+build and release process for the IBM CCM artifacts. The CCM command and artifact
+implementation should be straight-forward. As for build and release, IBM CCM will
+deliver version `X.Y.*` releases that are compatible with Kubernetes version
+`X.Y.*` releases. The patch version may not align. In general, branching,
+tagging and the release process will be similar to that of Kubernetes.
+
+IKS already has a CI/CD pipeline that runs conformance testing using both
+methods outlined in
+[KEP-0018-testgrid-conformance-e2e](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/0018-testgrid-conformance-e2e.md).
+This testing is done with IBM KCM and against every patch version of Kubernetes
+supported by IKS, which may include pre-release versions but does not include
+master. To close the remaining gaps, the IKS CI/CD pipeline will be updated to
+implement following requirements:
+- Run conformance testing using IBM CCM
+- Report conformance test results to [Testgrid](https://github.com/kubernetes/test-infra/tree/master/testgrid)
+- Ensure every patch version of Kubernetes is tested independent of IKS support
+
+The current [IKS documentation](https://cloud.ibm.com/docs/containers) implements
+most of the load balancer service documentation requirements per
+[KEP-20180731-cloud-provider-docs](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20180731-cloud-provider-docs.md).
+The remaining gaps must be filled and included within the IBM cloud provider
+repository. This will include IBM cloud provider documentation for the following:
+- DevOps
+- Service annotations, labels, etc.
+- Node annotations, labels, etc.
+
+Migrating existing clusters from IBM KCM to IBM CCM will follow the recommended
+process developed by the cloud provider SIG.
```

Comment:
Suggest to use `Cloud Provider SIG` with a hyperlink https://github.com/kubernetes/community/tree/master/sig-cloud-provider

Reply:
Done